src/master/x11.rs

```diff
@@ -1,68 +1,48 @@
+use crate::{CallbackResult, ClipboardHandler, Master};
 use std::io;
-use crate::{ClipboardHandler, CallbackResult, Master};
-
-use x11_clipboard::xcb;
 
 impl<H: ClipboardHandler> Master<H> {
     ///Starts Master by waiting for any change
     pub fn run(&mut self) -> io::Result<()> {
-        let mut result = Ok(());
-        let clipboard = x11_clipboard::Clipboard::new().unwrap();
-
-        let xfixes = match xcb::query_extension(&clipboard.getter.connection, "XFIXES").get_reply() {
-            Ok(xfixes) => xfixes,
-            Err(error) => return Err(io::Error::new(io::ErrorKind::Other, error)),
-        };
-        assert!(xfixes.present());
-        xcb::xfixes::query_version(&clipboard.getter.connection, 5, 0);
+        let clipboard = x11_clipboard::Clipboard::new();
+        if let Err(error) = clipboard {
+            return Err(io::Error::new(
+                io::ErrorKind::Other,
+                format!("Failed to initialize clipboard: {:?}", error),
+            ));
+        }
+        let clipboard = clipboard.unwrap();
 
         loop {
-            let selection = clipboard.getter.atoms.clipboard;
-
-            let screen = match clipboard.getter.connection.get_setup().roots().nth(clipboard.getter.screen as usize) {
-                Some(screen) => screen,
-                None => match self.handler.on_clipboard_error(io::Error::new(io::ErrorKind::Other, "Screen is not available")) {
+            let res = clipboard.load_wait(
+                clipboard.getter.atoms.clipboard,
+                clipboard.getter.atoms.utf8_string,
```

Comment:
will it only await on changes for string read/write operations?
What about other types of changes?

Reply:
Just test with images, and files, at least will await changes for these two.

Reply:
Hmm I see, that's surprising to know.
If it works with images and file copies, then it should be ok

Reply:
Also, just test with replace `clipboard.getter.atoms.utf8_string` to `clipboard.getter.atoms.incr`, also work.
Personally, `incr` maybe better, since the `load_wait` function will load the whole string into memory.

Reply:
`load_wait` loading whole content will probably clear content, right?
if `incr` prevents that, that would be better

Please make sure you can load content afterwards from user's callback https://github.com/DoumanAsh/clipboard-master/blob/master/src/lib.rs#L39

Reply:
As far as I tested now, I can still pasta the content directly and also get the content using [this](https://docs.rs/x11-clipboard/0.8.1/x11_clipboard/struct.Clipboard.html#method.load) load function.

Reply:
Sounds good then